modules/oidcsso/src/test/java/org/wso2/carbon/tomcat/oidcsso/extension/utils/OIDCConfigurationTest.java

```diff
@@ -0,0 +1,206 @@
+        oidcConfiguration.setLogoutEndpoint(URI.create(TestConstants.LOGOUT_ENDPOINT));
+
+        OIDCConfiguration.Truststore truststore = new OIDCConfiguration.Truststore();
+        truststore.setLocation(TestConstants.TRUSTSTORE_PATH);
+        truststore.setType(TestConstants.TYPE);
+        truststore.setPassword(TestConstants.TRUSTSTORE_PASSWORD);
+        truststore.setKeyAlias(TestConstants.PRIVATE_KEY_ALIAS);
+        oidcConfiguration.setTruststore(truststore);
+        truststore.setLocation(string_sub.replace(oidcConfiguration.getTruststore().getLocation()));
+        truststore.setLocation(StrSubstitutor
+                .replaceSystemProperties(oidcConfiguration.getTruststore().getLocation()));
+        return oidcConfiguration;
+    }
+
+    private static boolean compare(OIDCConfiguration actual, OIDCConfiguration expected) {
+        if ((actual != null) && (expected != null)) {
+            boolean enable = actual.isEnabled().equals(expected.isEnabled());
+            boolean clientID = actual.getClientID().trim().equals(expected.getClientID());
+            boolean clientSecret = actual.getClientSecret().trim().equals(expected.getClientSecret());
+            boolean redirectURI = actual.getRedirectURI().equals(expected.getRedirectURI());
+            boolean scope = actual.getScope().trim().equals(expected.getScope());
+            boolean claims = actual.getClaims().trim().equals(expected.getClaims());
+            boolean responseType = actual.getResponseType().trim().equals(expected.getResponseType());
+            boolean grantType = actual.getGrantType().trim().equals(expected.getGrantType());
+            boolean authenticationEndpoint = actual.getAuthenticationEndpoint()
+                    .equals(expected.getAuthenticationEndpoint());
+            boolean tokenEndpoint = actual.getTokenEndpoint().equals(expected.getTokenEndpoint());
+            boolean userInfoEndpoint = actual.getUserInfoEndpoint().equals(expected.getUserInfoEndpoint());
+            boolean logoutEndpoint = actual.getLogoutEndpoint().equals(expected.getLogoutEndpoint());
+            boolean security = comparetrustStoreConfigurations(actual.getTruststore(),
+                    expected.getTruststore());
+
+            return (enable && clientID && clientSecret && redirectURI && scope && claims && responseType && grantType
+                    && authenticationEndpoint && tokenEndpoint && userInfoEndpoint && logoutEndpoint && security);
+        } else {
+            return ((actual == null) && (expected == null));
+        }
+    }
+
+    private static boolean comparetrustStoreConfigurations(OIDCConfiguration.Truststore actual,
+                                                           OIDCConfiguration.Truststore expected) {
+        if ((actual != null) && (expected != null)) {
+            boolean truststorePath = actual.getLocation().trim().
+                    equals(expected.getLocation());
```

Comment:
Can you check whether we can move this line up? If it is possible, do the same to the next variable initialization.